pkg/storegateway/series_refs.go

```diff
@@ -898,17 +902,27 @@ func encodeCachedSeriesForPostings(set seriesChunkRefsSet, matchersKey indexcach
 	return snappy.Encode(nil, uncompressed), nil
 }
 
-func logSeriesForPostingsCacheEvent(ctx context.Context, logger log.Logger, userID string, blockID ulid.ULID, matchers []*labels.Matcher, shard *sharding.ShardSelector, postingsKey indexcache.PostingsKey, msgAndArgs ...any) {
+func logSeriesForPostingsCacheEvent(ctx context.Context, logger log.Logger, userID string, blockID ulid.ULID, shard *sharding.ShardSelector, itemID cachedSeriesForPostingsID, msgAndArgs ...any) {
 	nonNilShard := maybeNilShard(shard)
-	var matchersStr strings.Builder
-	for _, m := range matchers {
-		matchersStr.WriteString(m.String())
-		matchersStr.WriteString(",")
-	}
-	msgAndArgs = append(msgAndArgs, "tenant_id", userID, "block_ulid", blockID.String(), "matchers", matchersStr.String(), "requested_shard_index", nonNilShard.ShardIndex, "requested_shard_count", nonNilShard.ShardCount, "postings_key", postingsKey)
+	msgAndArgs = append(msgAndArgs,
+		"tenant_id", userID,
+		"block_ulid", blockID.String(),
+		"requested_shard_index", nonNilShard.ShardIndex,
+		"requested_shard_count", nonNilShard.ShardCount,
+		"postings_key", itemID.postingsKey,
+		"trimmed_postings", printDiffEncodedPostings(itemID.encodedPostings),
+	)
 	level.Warn(spanlogger.FromContext(ctx, logger)).Log(msgAndArgs...)
 }
 
+// printDiffEncodedPostings truncates the postings to just 100 bytes and prints them as a golang byte slice
+func printDiffEncodedPostings(b []byte) string {
```

Comment:
[nit] Since this function doesn't print it, I would just name it something like `previewDiffEncodedPostings()` (given it truncates to first 100 bytes).